tests/osquery/test_api_views.py

```diff
@@ -2401,3 +2764,283 @@ def test_delete_query_permission_denied(self):
         query = self.force_query()
         response = self.delete(reverse("osquery_api:query", args=(query.pk,)))
         self.assertEqual(response.status_code, 403)
+
+    # List ConfigurationPacks
+
+    def test_list_configurationpacks_unauthorized(self):
+        response = self.get(reverse("osquery_api:configuration_packs"), include_token=False)
+        self.assertEqual(response.status_code, 401)
+
+    def test_list_configurationpacks_permission_denied(self):
+        response = self.get(reverse("osquery_api:configuration_packs"))
+        self.assertEqual(response.status_code, 403)
+
+    def test_list_configurationpacks_filter_configuration_id_not_found(self):
+        self.set_permissions("osquery.view_configurationpack")
+        response = self.get(reverse("osquery_api:configuration_packs"), {"configuration_id": 9999})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [])
+
+    def test_list_configurationpacks_filter_pack_id_not_found(self):
+        self.set_permissions("osquery.view_configurationpack")
+        response = self.get(reverse("osquery_api:configuration_packs"), {"pack_id": 9999})
```

Comment:
Why are we not getting a `Select a valid choice. That choice is not one of the available choices.` like in the other part of the API?

Reply:
this is fixed, i am using ModelChoiceFilter in the api view